packages/engine/solana/collectibles.ts

```diff
@@ -0,0 +1,99 @@
+import {
+	type JsonMetadata,
+	type Metadata,
+	type Nft,
+	Metaplex,
+} from '@metaplex-foundation/js';
+import { type Connection, PublicKey } from '@solana/web3.js';
+import { type Endpoint, Networks } from '@walless/core';
+import {
+	type CollectibleDocument,
+	type CollectionDocument,
+} from '@walless/store';
+
+interface CollectiblesByAddressOptions {
+	endpoint: Endpoint;
+	connection: Connection;
+	address: string;
+}
+
+export const solanaCollectiblesByAddress = async ({
+	endpoint,
+	connection,
+	address,
+}: CollectiblesByAddressOptions): Promise<{
+	collections: CollectionDocument[];
+	collectibles: CollectibleDocument[];
+}> => {
+	const mpl = new Metaplex(connection);
+
+	const collections: CollectionDocument[] = [];
+	const collectibles: CollectibleDocument[] = [];
+
+	const nfts = await Promise.all(
+		(
+			await mpl.nfts().findAllByOwner({ owner: new PublicKey(address) })
+		).map((metadata) =>
+			mpl.nfts().load({ metadata: metadata as Metadata<JsonMetadata<string>> }),
+		),
+	);
+
+	const promises = nfts
+		.filter((ele) => ele.json)
+		.map(async (nft) => {
+			let collectionMetadata: Nft | undefined = undefined;
+
+			if (nft.collection) {
+				collectionMetadata = (await mpl
+					.nfts()
+					.findByMint({ mintAddress: nft.collection.address })) as Nft;
+			}
+
+			const collectionId = `${address}/${
+				collectionMetadata?.address.toString() ||
+				nft.collection?.address.toString() ||
+				nft.address.toString()
+			}`;
+
+			if (!collections.find((ele) => ele._id === collectionId)) {
```

Comment:
`find` could be replaced by `some` which return `boolean` type and give a more transparent idea about what condition we want to check.